Style changes and additional component for overview page

msecret opened the pull request:
Adds rules for more complex, nested panel rows and a new component for colorized app statuses.

View commit messages for details

Commit:
Dealing with complex, nested panel rows.

The overview page UI has rows within rows within rows, so added new
rules to accomodate this. It was easiest to use the > selector for all
immediately nested roles and then use one more of these selectors for
the 3rd or last nested row in this case. If theres a use case in the
future for a 4th nested row (probably shouldn't be) this code can
change then.

Also added TODOs to a bunch of the rules under panel-row, hoping they
can be moved elsewhere to reduce complexity of that component.

Commit:
New component for colored status text

I wasn't sure the best way to color code status text on the dashboard
so created a new component for it. It definitely doesn't fit anywhere
else, but does share some functionality similar to count_status.

I decided to make it similar to count_status to make things consistent
even though it could maybe do a better job of sharing things.

Commit:
Icons for CF entities

Such as space, app, org. Includes code that makes it easier to inline
icons rather then the block layout as done on cg-site. To inline an
icon, which is done on dashboard overview page, simply don't use the
icon container and place the icon within the text tag it's next to.

The inlined icon will keep the size of the text tag it's in as it's
height and width are set to 1em. The icon also has a min width/height
though as I found at too small of sizes of text the icon wasn't
rendering very well.

Some of these icons need additional work to receive the border they
have.

This PR addresses separating out fill icons vs regular (or existing)
stroke icons. This has to be done because some of our icons only work
as fills and vice-versa for strokes. The default is just the `icon`
class and is stroke. The `icon-fill` class is for fill icons and will
be colored differently.

adborden commented on src/css/components/panel.scss:
```diff
@@ -82,19 +82,42 @@ h4 + .panel {
     width: 100%;
   }
 
+  // TODO Where does these sizes come from? Move to typography somehow?
```
Let's create class based rules for these instead.

adborden commented on src/css/components/panel.scss:
```diff
@@ -82,19 +82,42 @@ h4 + .panel {
     text-decoration: underline;
   }
 
+  // TODO Does this need to be under .panel-row? It can probably be outside.
```
:+1:

adborden commented on src/css/components/status.scss:
```diff
@@ -0,0 +1,25 @@
+.status-error {
+  color: $color-error;
+
+}
```
Want to add components for these in the pattern library?

Commit:
Refactor panel row typography

And moved it to base typography where it makes more sense. To do this,
I had to create new sans classes for each of the sans type scale
sizes as certain of these sizes didn't exist in any place. These can
be used to scale type outside of the h tags.

Commit:
Document status component

Provide information on how to use the status components along with it's
various modifiers.

Commit:
Merge pull request #203 from 18F/ms-entity_icons

Icons for CF entities

Commit:
Add styles for count status parts

To separate the icon from the text of the count status.
